qt/scientific_interfaces/ISISReflectometry/GUI/Instrument/InstrumentPresenter.cpp

```diff
@@ -42,6 +44,20 @@ void InstrumentPresenter::notifyRestoreDefaultsRequested() {
   restoreDefaults();
 }
 
+void InstrumentPresenter::notifyEditingCalibFilePathFinished() {
+  auto const calibrationFilePath = m_view->getCalibrationFilePath();
```

Comment:
There's an interface `IReflMessageHandler` with a method `askUserForLoadFileName` you can use for getting the filename. (It's a work in progress but I've been trying to refactor out direct uses of QFileDialog.) It will require a bit of boilerplate to pass it in here but I think it's worth it. It can also be used for pop-up messages, so we can avoid direct use of QMessageBox too.